osu.Game.Rulesets.Osu/Edit/OsuSliderDrawingSettingsProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Utils;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Edit;
+
+namespace osu.Game.Rulesets.Osu.Edit
+{
+    public partial class OsuSliderDrawingSettingsProvider : Drawable, ISliderDrawingSettingsProvider
+    {
+        public BindableFloat Tolerance { get; } = new BindableFloat(0.1f)
+        {
+            MinValue = 0.05f,
+            MaxValue = 1f,
+            Precision = 0.01f
+        };
+
+        private readonly BindableInt sliderTolerance = new BindableInt(10)
+        {
+            MinValue = 5,
+            MaxValue = 100
+        };
+
+        private ExpandableSlider<int> toleranceSlider = null!;
+
+        public OsuSliderDrawingSettingsProvider()
+        {
+            sliderTolerance.BindValueChanged(v =>
+            {
+                float newValue = v.NewValue / 100f;
+                if (!Precision.AlmostEquals(newValue, Tolerance.Value, 1e-7f))
+                    Tolerance.Value = newValue;
+            });
+            Tolerance.BindValueChanged(v =>
+            {
+                int newValue = (int)Math.Round(v.NewValue * 100f);
+                if (sliderTolerance.Value != newValue)
+                    sliderTolerance.Value = newValue;
+            });
+        }
+
+        public void AttachToToolbox(ExpandingToolboxContainer toolboxContainer)
```

Comment:
We probably want to make a `IToolboxAttachment` interface out of this method or something given it's showed up on a second class now. But maybe as a follow-up PR.